zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1487,413 @@ where
+
+        let response = rpc_client.send_transaction(request).await?.into_inner();
+
+        assert_eq!(response, expected_response);
+    }
+
+    Ok(())
+}
+
+/// Loads transactions from a block that's after the chain tip of the cached state.
+///
+/// This copies the cached state into a temporary directory when it is needed to avoid overwriting
+/// anything. Two copies are made of the cached state.
+///
+/// The first copy is used by a zebrad instance connected to the network that finishes
+/// synchronizing the chain. The transactions are loaded from this updated state.
+///
+/// The second copy of the state is returned together with the transactions. This means that the
+/// returned tuple contains the temporary directory with the partially synchronized chain, and a
+/// list of valid transactions that are not in any of the blocks present in that partially
+/// synchronized chain.
+async fn load_transactions_from_a_future_block(
+    network: Network,
+    cached_state_path: PathBuf,
+) -> Result<(Vec<Arc<Transaction>>, TempDir)> {
+    let (partial_sync_path, partial_sync_height) =
+        prepare_partial_sync(network, cached_state_path).await?;
+
+    let full_sync_path =
+        perform_full_sync_starting_from(network, partial_sync_path.as_ref()).await?;
+
+    let transactions =
+        load_transactions_from_block_after(partial_sync_height, network, full_sync_path.as_ref())
+            .await?;
+
+    Ok((transactions, partial_sync_path))
+}
+
+/// Prepares the temporary directory of the partially synchronized chain.
+///
+/// Returns a temporary directory that can be used by a Zebra instance, as well as the chain tip
+/// height of the partially synchronized chain.
+async fn prepare_partial_sync(
+    network: Network,
+    cached_zebra_state: PathBuf,
+) -> Result<(TempDir, block::Height)> {
+    let partial_sync_path = copy_state_directory(cached_zebra_state).await?;
+    let partial_sync_state_dir = partial_sync_path.as_ref().join("state");
+    let tip_height = load_tip_height_from_state_directory(network, &partial_sync_state_dir).await?;
+
+    Ok((partial_sync_path, tip_height))
+}
+
+/// Loads the chain tip height from the state stored in a specified directory.
+async fn load_tip_height_from_state_directory(
+    network: Network,
+    state_path: &Path,
+) -> Result<block::Height> {
+    let (_state_service, _read_state_service, latest_chain_tip, _chain_tip_change) =
+        start_state_service(network, state_path).await?;
+
+    let chain_tip_height = latest_chain_tip
+        .best_tip_height()
+        .ok_or_else(|| eyre!("State directory doesn't have a chain tip block"))?;
+
+    Ok(chain_tip_height)
+}
+
+/// Runs a zebrad instance to synchronize the chain to the network tip.
+///
+/// The zebrad instance is executed on a copy of the partially synchronized chain state. This copy
+/// is returned afterwards, containing the fully synchronized chain state.
+async fn perform_full_sync_starting_from(
```

Comment:
Optional: We could move this utility function to one of the test sub-modules.

Reply:
Done.